pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -787,7 +788,16 @@ impl Timeline {
                 //
                 // Suppress error when it's due to cancellation
                 if !self.cancel.is_cancelled() && !err.is_cancelled() {
-                    tracing::error!("could not compact, repartitioning keyspace failed: {err:?}");
+                    if let CompactionError::CollectKeySpaceError(
+                        CollectKeySpaceError::Decode(_) | CollectKeySpaceError::PageRead(_),
```

Comment:
I think there's still `CollectKeySpaceError` variants here we don't care about, e.g. `AncestorLsnTimeout`.

I'll let you judge how fine-grained this needs to be, just note that this will page on-call when it fires.